code/modules/lighting/lighting_source.dm

```diff
@@ -60,7 +60,8 @@
 	if (needs_update)
 		GLOB.lighting_update_lights -= src
 
-	. = ..()
+	..()
+	return QDEL_HINT_IWILLGC
```

Comment:
There is a likelihood of failures with this datum since its referenced by alot of intermingling datums